apis/v1alpha2/validation/gateway_test.go

```diff
@@ -141,6 +156,44 @@ func TestValidateGateway(t *testing.T) {
 			},
 			expectErrsOnFields: []string{"spec.listeners[0].hostname"},
 		},
```

Comment:
add tests and validation code for:
- Protocol is TLS but TLSConfig is not specified - failure
- Protocol is HTTPS but TLSConfig is not specified - failure

Reply:
I don't think we can safely validate that without conflicting with cert generation or other non-secret based certificates. In at least some cases users will want to leave this blank initially and have it filled in later with automation (or connected to some other kind of certificate).

Reply:
Don't you need at least the `Mode` bit set?

Reply:
There's a default value of `Terminate` for that field which seems sufficient to me

Reply:
You are right. Resolving the thread.